lib/propolis/src/hw/virtio/queue.rs

```diff
@@ -770,3 +770,55 @@ pub(crate) fn write_buf(buf: &[u8], chain: &mut Chain, mem: &MemCtx) {
         }
     });
 }
+
+#[cfg(feature = "falcon")]
+pub(crate) fn p9_write_file(
+    file: &impl std::os::fd::AsRawFd,
+    chain: &mut Chain,
+    mem: &MemCtx,
+    count: usize,
+    offset: i64,
+) {
```

Comment:
Is there a reason why this can't go in `p9fs.rs`?

Reply:
Nope, I put it here as the `write_buf` function is here which is very similar. I can move it to `p9fs.rs`

Reply:
Since it's p9-specific, I'd keep it with the rest of the p9-related things